src/main/com/vu/psp/evaldasg/validators/EmailValidator.java

```diff
@@ -0,0 +1,205 @@
+                if (parts.size() > 8)
+                    return false;
+                if (isDigit(current) || isHexLetter(current)) {
+                    temp += current;
+                } else {
+                    return false;
+                }
+            }
+            parts.add(temp);
+            if (parts.contains("") || (hasDoubleColons > 1) || (hasDoubleColons == 0 && parts.size() != 8))
+                return false;
+        } else {
+            String internationalizedDomain = domain.substring(1, domainLength - 1); // [192.168.2.1]
+            String temp = "";
+            List<String> parts = new ArrayList<>();
+            int counter = 0;
+            for (int i = 0; i < internationalizedDomain.length(); i++) {
+                char current = internationalizedDomain.charAt(i);
+                counter++;
+                if (current == '.') {
+                    parts.add(temp);
+                    temp = "";
+                    counter = 0;
+                    continue;
+                }
+                if (counter > 3)
+                    return false;
+                if (parts.size() > 4)
+                    return false;
+                if (isDigit(current)) {
+                    temp += current;
+                    if (Integer.parseInt(temp) > 255)
+                        return false;
+                } else {
+                    return false;
+                }
+            }
+        }
+        return validDomains.contains(domain);
+    }
+
+    private String getLp(String email) {
+        return email.substring(0, email.indexOf("@"));
+    }
+
+    private String getDomain(String email) {
+        return email.substring(email.indexOf("@") + 1);
+    }
+
+    private boolean validateLp(String lp) {
+        int lpLength = lp.length();
+        if (lpLength == 0 || lpLength > 64)
+            return false;
+        if (!(lp.charAt(0) != '.' && lp.charAt(lpLength - 1) != '.'))
+            return false;
+        if (lpContainsMultipleDots(lp, lpLength))
+            return false;
+        return validateLpSymbols(lp, lpLength);
+    }
+
+    private boolean lpContainsMultipleDots(String lp, int lpLength) {
+        if (lp.contains("."))
+            for (int i = lp.indexOf("."); i < lpLength; i++)
+                if ((i + 1) != lpLength && lp.charAt(i) == '.' && lp.charAt(i + 1) == '.')
+                    return true;
+        return false;
+    }
+
+    private boolean validateLpSymbols(String lp, int lpLength) {
+        for (int i = 0; i < lpLength; i++) {
+            if (!isSymbolCorrectInLp(lp.charAt(i))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean isSymbolCorrectInDomain(char domain) {
+        return domain == 91 || domain == 93 || domain == 58 ||
+                (domain >= 97 && domain <= 122) ||
+                (domain >= 48 && domain <= 57) ||
+                domain == 45 || domain == 46;
+    }
```

Comment:
The same idea as in below isSymbolCorrectInLp (Lp - local-part) method but with an inconvenient way for readability.

Reply:
Pletkininkas === Evaldas